ft_split.c

```diff
@@ -0,0 +1,55 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   ft_split.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: bchristo <bchristo@student.42wolfsburg.    +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2021/05/26 21:10:01 by bchristo          #+#    #+#             */
+/*   Updated: 2021/05/27 03:37:22 by bchristo         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "libft.h"
+
+static int	ft_isspace(int c)
+{
+	if (c == '\t' || c == '\n' || c == '\v' || c == '\f' || c == '\r' || c == ' ')
+		return (1);
+	return (0);
+}
+
+static void	init(char **r, size_t *i)
+{
+	r[0] = NULL;
+	*i = 0;
+}
+
+char	**ft_split(char const *s, char c)
+{
+	char		**r;
+	char		*end;
+	size_t		i;
+
+	r = (char **)malloc((ft_strlen(s) + 1) * sizeof(char *));
```

Comment:
This allocates more memory than needed. It won't cause problem, but it is not a nice practice.

Reply:
if I remove it everything breaks. I think this extra +1 is there for null terminating the string?